### PR TITLE
fix: unquote instance name

### LINKED ISSUE #146: [Bug] Conflicting names value when using quote for instance name
When using `"pod-info"` as an instance name, the build fails.

```cue
bundle: {
    apiVersion: "v1alpha1"
    name:       "podinfo"
    instances: {
        redis: {
            module: {
                url:     "oci://ghcr.io/stefanprodan/modules/redis"
                version: "7.0.10"
            }
            namespace: "podinfo"
            values: maxmemory: 256
        }
        "pod-info": {
            module: url:     "oci://ghcr.io/stefanprodan/modules/podinfo"
            module: version: "6.3.5"
            namespace: "podinfo"
            values: caching: {
                enabled:  true
                redisURL: "tcp://redis:6379"
            }
        }
    }
}
```

```shell
> timoni bundle build -f podinfo.bundle.cue
12:15PM ERR failed to build instance:
timoni.instance.config.metadata.name: 2 errors in empty disjunction:
timoni.instance.config.metadata.name: conflicting values "podinfo" and "\"pod-info\"":
    ./templates/config.cue:14:24
    ./templates/config.cue:81:10
    ./timoni.cue:24:12
timoni.instance.config.metadata.name: invalid value "\"pod-info\"" (out of bound =~"^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?$"):
    ./templates/config.cue:14:45
    ./cue.mod/gen/k8s.io/apimachinery/pkg/apis/meta/v1/types_go_gen.cue:97:9
    ./templates/config.cue:14:23
    ./timoni.cue:31:15
```

At first sight it seems a double escape is done somewhere `"\"pod-info\""`. I'll investigate more and try to push a fix. Also not sure if it's a regression or if it was already there as I only tried with the latest version of timoni.

---

```shell
timoni version
api: timoni.sh/v1alpha1
client: 0.0.0-fe12735.1687375127
cue: 0.5.0
```

### PR DESCRIPTION
**Pull request by b4nst**
Fixes #146

### CODE REVIEW

**Review by stefanprodan — COMMENTED** [review]
Can you please add a test to prove the bug is fixed.

**Comment by b4nst** [comment]
Done

File: internal/engine/bundle_builder_test.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            module: url:     "oci://ghcr.io/stefanprodan/modules/podinfo"
+            module: version: "6.3.5"
+            namespace: "podinfo"
+            values: caching: {
+                enabled:  true
+                redisURL: "tcp://redis:6379"
+            }
+        }
+    }
+}
```
Comment (stefanprodan) [inline internal/engine/bundle_builder_test.go:46]:
```suggestion
    instances: {
        "pod-info": {
            module: url:     "oci://ghcr.io/stefanprodan/modules/podinfo"
            module: version: "6.3.5"
            namespace: "podinfo"
            values: caching: {
                enabled:  true
                redisURL: "tcp://redis:6379"
            },
         podinfo: {
            module: url:     "oci://ghcr.io/stefanprodan/modules/podinfo"
            module: version: "6.3.5"
            namespace: "podinfo"
            values: caching: {
                enabled:  true
                redisURL: "tcp://redis:6379"
            }
        }
    }
}
```

Let's add the other variant as well, to test that both are working.

Comment (stefanprodan) [reply internal/engine/bundle_builder_test.go]: this will not work as it is, the test needs adjustments

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @b4nst for the fix!

**Pull request merged**